### PR TITLE
Added retrieveUserById to ShardManager

### PR DESCRIPTION
**Pull request by Artuto**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

There are several guidelines you should follow in order for your
  Pull Request to be merged.

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

> It is sometimes better to include more changes in a single commit. 
  If you find yourself having an overwhelming amount of commits, you
  can **rebase** your branch.

## Description

This PR adds the `retrieveUserById` method to the ShardManager.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java [hunk 1]
```diff
@@ -446,6 +451,85 @@ default Guild getGuildById(final String id)
...
+    default RestAction<User> retrieveUserById(long id)
+    {
+        // We need a JDA instance
+        JDA api = this.getShardCache().stream()
+                .findAny().orElseThrow(() -> new IllegalStateException("no active shards"));
+
+        // check cache
+        User user = this.getUserById(id);
+        if (user != null)
+            return new RestAction.EmptyRestAction<>(api, user);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java]: This should use the JDA instance of the user instead.

Comment (Artuto) [reply src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java]: Done

File: src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java [hunk 2]
```diff
@@ -446,6 +451,85 @@ default Guild getGuildById(final String id)
...
+     *
+     * @param  id
+     *         The id of the requested {@link net.dv8tion.jda.core.entities.User User}.
+     *
+     * @throws java.lang.NumberFormatException
+     *         If the provided {@code id} cannot be parsed by {@link Long#parseLong(String)}
+     * @throws java.lang.IllegalArgumentException
+     *         <ul>
+     *             <li>If the provided id String is null.</li>
+     *             <li>If the provided id String is empty.</li>
```
[the remark this answers is not in the document]
Comment (Artuto) [reply src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java]: So also remove the throw declaration of NumberFormatException?

**Comment by MinnDevelopment** [comment]
Waiting for review by @Almighty-Alpaca

**Pull request merged**